VideoLocker/src/main/java/org/edx/mobile/view/view_holders/AuthorLayoutViewHolder.java

```diff
@@ -1,16 +1,108 @@
+import android.support.annotation.NonNull;
+import android.support.v4.widget.TextViewCompat;
 import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
 import android.widget.TextView;
 
+import com.bumptech.glide.BitmapRequestBuilder;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.resource.drawable.GlideDrawable;
+import com.google.inject.Inject;
+import com.joanzapata.iconify.IconDrawable;
+import com.joanzapata.iconify.fonts.FontAwesomeIcons;
+
 import org.edx.mobile.R;
+import org.edx.mobile.discussion.DiscussionTextUtils;
+import org.edx.mobile.discussion.IAuthorData;
+import org.edx.mobile.model.api.ProfileModel;
+import org.edx.mobile.module.prefs.UserPrefs;
+import org.edx.mobile.user.ProfileImage;
+import org.edx.mobile.user.ProfileImageProvider;
+import org.edx.mobile.util.Config;
+
+import roboguice.RoboGuice;
 
 public class AuthorLayoutViewHolder {
+    @Inject
+    private UserPrefs userPrefs;
 
-    public final TextView discussionAuthorTextView;
+    public final ViewGroup profileRow;
+    public final ImageView profileImageView;
+    public final TextView authorTextView;
+    public final TextView dateTextView;
+    public final TextView answerTextView;
 
     public AuthorLayoutViewHolder(View itemView) {
-        discussionAuthorTextView = (TextView) itemView.
-                findViewById(R.id.discussion_author_layout_author_text_view);
+        profileRow = (ViewGroup) itemView;
+        profileImageView = (ImageView) itemView.findViewById(R.id.profile_image);
+        authorTextView = (TextView) itemView.findViewById(R.id.discussion_author_text_view);
+        dateTextView = (TextView) itemView.findViewById(R.id.discussion_date_text_view);
+        answerTextView = (TextView) itemView.findViewById(R.id.discussion_responses_answer_text_view);
+
+        final Context context = answerTextView.getContext();
+        TextViewCompat.setCompoundDrawablesRelativeWithIntrinsicBounds(
+                answerTextView,
+                new IconDrawable(context, FontAwesomeIcons.fa_check_square_o)
+                        .sizeRes(context, R.dimen.edx_xxx_small)
+                        .colorRes(context, R.color.edx_utility_success),
+                null, null, null);
+        RoboGuice.getInjector(context).injectMembers(this);
+    }
+
+    public void populateViewHolder(@NonNull Config config, @NonNull IAuthorData authorData,
+                                   @NonNull ProfileImageProvider provider,
+                                   long initialTimeStampMs,
+                                   @NonNull final Runnable listener) {
+        final Context context = profileImageView.getContext();
+        final ProfileImage profileImage;
+        {
+            if (provider.getProfileImage() != null) {
+                profileImage = provider.getProfileImage();
+            } else {
+                // TODO: Remove this else block when MA-2542 is fixed.
```

Comment:
We shouldn't need the profile image to be returned in the PATCH response since the thread should already have an image with it. Voting/reporting/editing does not affect the profile image so we shouldn't need to fetch that information again. Successful PATCHes should just update the appropriate UI instead of replacing the entire thread/comment/response.